### PR TITLE
feat: Add an "Invite" button to the workspace profile page

### PR DESCRIPTION
**Pull request by Krishna2323**
### Explanation of Change
- Adds an invite button beside `share` button on workspace profile page.

### Fixed Issues
$ https://github.com/Expensify/App/issues/53716
PROPOSAL: https://github.com/Expensify/App/issues/53716#issuecomment-2523951365

### Tests
1. Go settings > workspaces > select a workspace where you are admin
2. On profile page verify there is a invite button beside `share` button
3. Click on the invite button > verify you are taken to invite members page and members page is loaded in the background

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Go settings > workspaces > select a workspace where you are admin
2. On profile page verify there is a invite button beside `share` button
3. Click on the invite button > verify you are taken to invite members page and members page is loaded in the background

### QA Steps
1. Go settings > workspaces > select a workspace where you are admin
2. On profile page verify there is a invite button beside `share` button
3. Click on the invite button > verify you are taken to invite members page and members page is loaded in the background

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/3e4b612e-e771-4037-8a72-a3b7dab41617

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/43a638df-042d-4232-9e91-4159a5f11eae

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/52af2243-62cc-475c-b038-bad41b12a22e

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/31b37385-98fe-4d27-9374-1fc198923d5d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/b9e99d6d-49eb-463b-a629-aa354c0434e7

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/7a808e89-9e56-43d5-ba9c-004db4b21b1c

</details>

### CODE REVIEW

**Comment by neil-marcellini** [comment]
@Krishna2323 I don't really see any problems in those two videos that seem worth fixing. I think the navigation makes sense.

**Review by neil-marcellini — APPROVED** [review]
Looks great, assuming C+ testing passes

**Comment by dannymcclain** [comment]
Videos are looking good to me.

Regarding this:
>On small-width devices, when navigating to the invite page from the profile page, there is a flash that appears to be from the members page.

I actually think this might be a good thing because it might be easy to feel "lost" once you click the invite button, so showing a snippet of where the invite page is coming from is potentially a good thing. 🤷‍♂️

**Comment by shawnborton** [comment]
Yup, agree with those comments from Danny above, looking good!

**Review by allgandalf — APPROVED** [review]
LGTM!

**Review by allgandalf — APPROVED** [review]

**Comment by dubielzyk-expensify** [comment]
Happy to merge this, but seeing this now I actually think the RHP should fly out with the current page (Workspace profile) still underneath and only if you successfully add someone, land on members page highlighting the newly invited people. 

The jump from Invite button press to Members page feels a bit disorienting to me. 

Keen to hear @Expensify/design thoughts on that, but happy to move on still

**Comment by shawnborton** [comment]
I can totally see that Jon, happy to go with your recommendation as I can see where it might make things feel a bit more clear. 

The original intention was to "teach someone how to fish" by taking them first to the Members page but after seeing it in action I think I agree, the quick jump to a different page feels unexpected for a button like this.

**Comment by dannymcclain** [comment]
Yep, I [originally proposed](https://expensify.slack.com/archives/C03U7DCU4/p1733418813830879?thread_ts=1733359568.246879&cid=C03U7DCU4) exactly the flow you laid out for all the reasons that you mentioned, but others in the thread wanted to go the Members page route because of the "teach you to fish" idea that Shawn mentioned and because it was deemed as easier/faster to implement since the invite flow on the members page already exists.

TLDR: I'm very much for making the invite RHP open over the profile. We might want to check and see if the rest of the team is down with that change though.

**Review by neil-marcellini — DISMISSED** [review]
Good to go. If we decide to change the UX later we can do that in a follow up.

**Comment by shawnborton** [comment]
Hmm why a follow up? This feature doesn't exist yet so if we're going to change it, I feel like it makes sense to do so before merging this.

**Comment by neil-marcellini** [comment]
Ok that's fine too. It wasn't clear to me that the comments from the design team were blocking. Heads up, I'm going to be OOO for the rest of the week. I'm sure if you post in engineering chat someone can help review after the fix.

**Comment by dubielzyk-expensify** [comment]
If all of @Expensify/design is aligned, I recon we go with the change then:
1. Click Invite button
2. Invite member panel opens in RHP with Workspace Profile underneath
3. If dismissed, RHP goes away
4. If successfully invited someone, auto-navigate/land on Members screen with the newly added member highlighted

**Comment by shawnborton** [comment]
I say let's consider it decided - Jon has spoken - let's move forward with the changes 👍

**Comment by dannymcclain** [comment]
Agree. I think it's a much better experience that way.

**Review by neil-marcellini — APPROVED** [review]
Looks good to me

**Comment by izarutskaya** [comment]
@Krishna2323 QA found issue when validation this PR - issue https://github.com/Expensify/App/issues/54489

https://platform.applause.com/services/links/v1/external/af81e41beab72cb3ac2216bd2ea47a01f439ae0abd4dd7e1415dff60653f72f9
